### ISSUE TITLE
Make it a desktop application & add room passwords

### ISSUE THREAD

**Issue by cyberhardt** [howardchung/watchparty#763]
Compared to discord's screen sharing, Watchparty's is miles ahead and can transmit high quality files with practically no lag. the only issue is that, as a web page, it isn't able to, for example, share audio properly. While I was able to share a video on VLC and mpv, it lacked sound. It was also not able to transmit sound while sharing the entire screen (only the browser's sounds). As a firefox user, I wasn't able to upload it either. Perhaps, as a desktop application, either native or running on electron, it would be able to do what it already does and much more, while also being more performant and independent from specific browsers' limitations.

I believe it would be also nice to be able to set passwords for rooms.

**Comment by howardchung** [comment]
Passwords should already be supported!

Good point on the desktop application enabling more sharing--electron could be a good option to let us reuse most of the code.

**Comment by cyberhardt** [comment]
Are passwords supported? I couldn't find an option to set a password. Do I need to be logged in? If so, that's why I was not able to find it

Regarding electron, you could even make it in chromium directly. I'm not a big fan of them at all, but it is the uncomplicated, straightforward way to do it, ensuring as big of a compatibility across OS's as a web page on a regular browser

**Comment by howardchung** [comment]
Yes, logging in provides additional room options such as making it permanent, passwords, and kicking users

**Comment by evangineer** [comment]
An alternative to Electron worth considering is Tauri https://tauri.app/.
